src/sqllogictest/src/runner.rs

```diff
@@ -964,6 +968,10 @@ impl RunnerInner {
                 sql,
                 location,
             } => {
+                if self.auto_transactions && *in_transaction {
+                    self.client.execute("COMMIT", &[]).await?;
```

Comment:
So COMMIT is run before any `statement` directives, and BEGIN is run before any `query` directives? I'm pretty confused about the intentions here. If we decide we want this feature it needs many more comments about when and why we want to auto inject these txn controls.

Reply:
This is only done when `--auto-transactions` is set, which we only do for the automatically generated sqlite SLT tests that have been running way too slow to finish in 10 hours: https://buildkite.com/materialize/sql-logic-tests
It definitely doesn't work for all scenarios, but is fine for these specific tests.

Reply:
I'm more confused by WHEN we are injecting these statements. Is my read of the code accurate?

Reply:
`BEGIN` is inserted before a expected-successful query (`SELECT`), then we remember that we have an open transaction already, and don't run it on following statements.

`COMMIT` is inserted before an expected-failed statement, a `SHOW` and a statement (`CREATE/INSERT`).

So all blocks of successful `SELECT` end up in one transaction, everything else is outside of the explicit transactions.